File: Backend/auth-services/dto/auth_dto.go
package dto

type RegisterRequest struct {
	Email    string `json:"email" binding:"required,email"`
	Password string `json:"password" binding:"required,min=6"`
	Username string `json:"username" binding:"required,min=3"`
}

type LoginRequest struct {
	Email    string `json:"email" binding:"required,email"`
	Password string `json:"password" binding:"required,min=6"`
}

type RefreshTokenRequest struct {
	RefreshToken string `json:"refresh_token"`
}

type AuthResponse struct {
	AccessToken  string `json:"access_token"`
	RefreshToken string `json:"refresh_token"`
	UserID       string `json:"user_id"`
	Username     string `json:"username"`
	Email        string `json:"email"`
	Role         string `json:"role"`
	AvatarURL    string `json:"avatar_url"`
}

type ValidateTokenResponse struct {
	Valid    bool   `json:"valid"`
	Email    string `json:"email"`
	UserID   string `json:"user_id"`
	Username string `json:"username"`
	Role     string `json:"role"`
}

type MessageResponse struct {
	Message string `json:"message"`
}

type UserResponse struct {
	ID        string `json:"id"`
	Username  string `json:"username"`
	Email     string `json:"email"`
	Role      string `json:"role"`
	AvatarURL string `json:"avatar_url"`
}

type UpdateProfileRequest struct {
	Username string `json:"username" binding:"required,min=3"`
}

File: frontend/src/dto/authDto.ts
export interface LoginRequest {
    email: string;
    password: string;
}

export interface RegisterRequest {
    email: string;
    password: string;
    username: string;
}

export interface AuthResponse {
    access_token: string;
    refresh_token: string;
    user_id: string;
    username: string;
    email: string;
    role: string;
    avatar_url: string;
}

export interface MessageResponse {
    message: string;
}

export interface User {
    userId: string;
    role: string;
    email?: string;
    username?: string;
    avatarUrl?: string;
}

export interface UserResponse {
    id: string;
    username: string;
    email: string;
    role: string;
    avatar_url: string;
}

export interface UpdateProfileRequest {
    username: string;
}
